gsf-core/src/main/java/tools/gsf/navigation/siteplan/SitePlanNavService.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2016 Function1. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package tools.gsf.navigation.siteplan;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import COM.FutureTense.Interfaces.ICS;
+import com.fatwire.assetapi.data.AssetId;
+import com.fatwire.cs.core.db.PreparedStmt;
+import tools.gsf.facade.assetapi.AssetIdUtils;
+import tools.gsf.facade.assetapi.asset.TemplateAssetAccess;
+import tools.gsf.facade.runtag.render.LogDep;
+import tools.gsf.facade.sql.Row;
+import tools.gsf.facade.sql.SqlHelper;
+import tools.gsf.navigation.AssetNode;
+import tools.gsf.navigation.NavService;
+
+import java.util.*;
+
+/**
+ * Simple navigation service implementation that loads objects from the Site Plan.
+ * 
+ * Reads the full site plan tree in one query. It does not filter assets by site.
+ * 
+ * Nodes are instantiated via a dedicated method where you can load any data that
+ * is required; you get to (define and) use your own AssetNode implementation.
+ *
+ * 
+ * @author Tony Field
+ * @since 2016-07-06
+ */
+public abstract class SitePlanNavService<ANODE extends AssetNode<ANODE>> implements NavService<ANODE, AssetId, AssetId> {
```

Comment:
I used ANODE so to make it clear that it refers to an AssetNode, as opposed to NODE which refers to a Node